### PR TITLE
Support SUSE Manager on openSUSE

### PR DESCRIPTION
**Pull request by juliogonzalez**
Superseeds #272 (after the last sumaform changes I was afraid of having to resolve a lot of merge conflicts, so just in case...)

```main.tf``` example:

```
module "server" {
  source = "./modules/libvirt/suse_manager"
  base_configuration = "${module.base.configuration}"

  name = "server"
  version = "head"
  image = "opensuse423"
  // see modules/libvirt/suse_manager/variables.tf for possible values
}

module "proxy" {
  source = "./modules/libvirt/suse_manager_proxy"
  base_configuration = "${module.base.configuration}"
  image = "opensuse423"
  name = "proxy"
  version = "head"
  server_configuration = "${module.server.configuration}"
}

```

### CODE REVIEW

**Comment by juliogonzalez** [comment]
Do not aprove yet: I am fixing /etc/zypp/vendors.d/

**Comment by juliogonzalez** [comment]
Ping @moio: ready to be reviewed (note that this PR is much more complex than the original  #272, that can be closed).

File: salt/default/repos.sls [hunk 1]
```diff
@@ -303,12 +305,21 @@ tools_devel_repo:
 
 
 allow_vendor_changes:
+  {% if grains['osfullname'] == 'Leap' %}
+  file.managed:
+    - name: /etc/zypp/vendors.d/opensuse
```
Comment (moio) [inline salt/default/repos.sls]: Is the file name the only difference here, is it necessary for the vendor mechanism to work?

If so please restrict the conditional to the `name` line.

File: salt/mirror/minima.yaml [hunk 2]
```diff
@@ -268,3 +278,8 @@
 - url: http://download.opensuse.org/repositories/security:/logging/SLE_12_SP2
   path: /srv/mirror/repositories/security:/logging/SLE_12_SP2
   archs: [x86_64]
+
+# systemsmanagement:saltstack:products:next (salt for OpenSUSE 42.3)
```
Comment (moio) [inline salt/mirror/minima.yaml]: Nitpick: **o**penSUSE :wink:

File: salt/suse_manager_proxy/repos.d/Devel_Galaxy_Manager_Head_Leap.repo [hunk 3]
```diff
@@ -0,0 +1,6 @@
+[Devel_Galaxy_Manager_Head]
+name=Devel Project for SUSE Manager Head  (openSUSE Leap 42.3)
```
Comment (moio) [inline salt/suse_manager_proxy/repos.d/Devel_Galaxy_Manager_Head_Leap.repo]: Typo: double space :wink:

File: salt/suse_manager_proxy/repos.d/Devel_Galaxy_Manager_Head_Leap.repo [hunk 4]
```diff
@@ -0,0 +1,6 @@
+[Devel_Galaxy_Manager_Head]
```
Comment (moio) [inline salt/suse_manager_proxy/repos.d/Devel_Galaxy_Manager_Head_Leap.repo]: Unwritten convention here is to have the repo name equal to the `.repo` filename. Please uniform the two, and decide if we want to always prepend or always append `_Leap`.

I am fine with any as long as it is consistent.

File: salt/suse_manager_proxy/repos.d/Leap-SUSE-Manager-Proxy-Head-x86_64-Pool.repo [hunk 5]
```diff
@@ -0,0 +1,6 @@
+[SUSE-Manager-Proxy-Head-x86_64-Pool]
```
Comment (moio) [inline salt/suse_manager_proxy/repos.d/Leap-SUSE-Manager-Proxy-Head-x86_64-Pool.repo]: See above

File: salt/suse_manager_server/repos.d/Devel_Galaxy_Manager_Head_Leap.repo [hunk 6]
```diff
@@ -0,0 +1,6 @@
+[Devel_Galaxy_Manager_Head]
+name=Devel Project for SUSE Manager Head  (openSUSE Leap 42.3)
```
Comment (moio) [inline salt/suse_manager_server/repos.d/Devel_Galaxy_Manager_Head_Leap.repo]: Typo: double space :wink:

File: salt/default/repos.sls (hunk 1 above)
Comment (juliogonzalez) [reply salt/default/repos.sls]: Not just the filename, but the contents as well!

**Comment by moio** [comment]
Amazing, thanks!

**Pull request merged**